### ISSUE TITLE
[ops]: ttnn.geglu hardcoded the fast-LUT GELU gate (max 62571 ULP, negative tail truncated to zero)

### ISSUE THREAD

**Issue by abdulla-TT** [tenstorrent/tt-metal#54903]
### Component / Area

ops

### Issue Type (optional)

Bad Outputs

### Observed

`ttnn.geglu` computes `A * gelu(B)`, but the gate call hardcodes the fast approximate GELU:

```cpp
// ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_composite_op.cpp
constexpr bool fast_appx = true;
Tensor gelu_b = ttnn::gelu(ab[1], fast_appx, output_mem_config);
```

That `true` selects `GeluVariant::FAST_LUT`, a 6-segment piecewise-linear LUT. Every `ttnn.geglu` caller silently gets the least accurate GELU available, and there is no argument to opt out.

Exhaustive bfloat16 characterisation over all 65,026 non-subnormal gate values measures **max 62571 ULP**, with 49.7% of inputs above 1 ULP and 45.0% above 100 ULP.

Two structural failure modes, both measured on Wormhole n150:

| Gate B | `ttnn.geglu` today | `torch.nn.functional.gelu` | Error |
| --- | --- | --- | --- |
| -4.0 | `0` | -1.26719e-4 | 100% (sign lost) |
| -3.5 | `0` | -8.14229e-4 | 100% (sign lost) |
| 0.25 | 0.172852 | 0.149677 | 15.5% relative |
| 2.5 | 2.46875 | 2.48448 | 0.6% relative |

The negative tail collapsing to exactly zero is inherent to the LUT's structure: it evaluates `0.5x + f(|x|)`, and because the table is indexed on `|x|`, the `|x| >= 3` segment must serve both signs. Positive `x` forces that segment to `(slope 0.5, intercept 0)` so that `g = x`, and the same line makes the negative tail cancel to exactly 0 where the true GELU is small but nonzero.

### Expected

`ttnn.geglu` should default to the accurate GELU gate.

`ttnn.gelu` already has `GeluVariant::ACCURATE` — the piecewise-CDF (BF16) / FP32-erf kernel documented at Max ULP = 1. This is not a math problem; `geglu` simply is not reaching for it. The gate should be bitwise identical to `ttnn.gelu` with the corresponding variant, so `ttnn.gelu`'s own accuracy coverage carries over, and the fast path should remain reachable for callers who want it.

### 1. Steps (exact commands)

```bash
python /tmp/repro_geglu.py
```

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)

# A = 1.0 in the first half, gate value B in the second half, so geglu == gelu(B)
for b in (-4.0, -3.5, 2.5, 0.25):
    B = torch.full((1, 1, 32, 32), b, dtype=torch.bfloat16)
    A = torch.ones_like(B)
    x = ttnn.from_torch(
        torch.cat([A, B], dim=-1), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device
    )
    got = ttnn.to_torch(ttnn.geglu(x, dim=-1))[0, 0, 0, 0].item()
    ref = torch.nn.functional.gelu(torch.tensor(b, dtype=torch.float32)).item()
    print(f"B={b:>6}  ttnn.geglu={got:>12.6g}  torch={ref:>12.6g}")

ttnn.close_device(device)
```

Output on an unpatched tree:

```
B=  -4.0  ttnn.geglu=           0  torch=-0.000126719
B=  -3.5  ttnn.geglu=           0  torch=-0.000814229
B=   2.5  ttnn.geglu=     2.46875  torch=     2.48448
B=  0.25  ttnn.geglu=    0.172852  torch=     0.149677
```

For the full exhaustive sweep, run the bfloat16 ULP characterisation over all 65,536 patterns with `A` fixed to 1.0 to isolate the gate.

### 2. Input data / link or description

None. The repro is self-contained and needs no external data — `A = 1.0` makes the multiply exact so the output is the gate value itself.

### 3. Frequency

Always. Deterministic, not input-dependent in the sense of flakiness: every call to `ttnn.geglu` takes the fast-LUT path.

### 1. Software Versions

```
OS version: Ubuntu 22.04.5 LTS
tt-metal commit: 74a403ed2ec
```

### 2. Hardware Details

```
Product: Wormhole
Card/System: n150 L
```

### Is this a regression?

No

### Regression Details

Not a regression. The hardcoded `fast_appx = true` has been present since `geglu` was written; it was never wired to the accurate variant.

Note that the existing test could not have caught this. `test_geglu_op` asserted PCC >= 0.999, which passes for *every* GELU variant including the fast LUT at 62571 ULP, so the gate implementation was effectively unpinned.

### Logs & Diagnostics

Exhaustive bfloat16 ULP distribution for the gate, all 65,536 patterns (N = 65,026 non-subnormal):

```
--- geglu — all patterns (non-subnormal inputs) | N=65,026 ---
  Max ULP     : 62571
  ULP > 1     : 32,328  (49.7155%)
  ULP = 0     : 32,460  (49.9185%)
  ULP = 1     : 238  (0.3660%)
  ULP = 2     : 81  (0.1246%)
  ULP 3-10    : 257  (0.3952%)
  ULP 11-100  : 2,730  (4.1983%)
  ULP > 100   : 29,260  (44.9974%)
```

### Priority

P2

### Impact

Affects every `ttnn.geglu` caller. GEGLU is a standard transformer feed-forward activation, so any model using it silently accumulates gate error, with the worst case being the negative tail truncated to exactly zero.

Fixing the default is a behaviour change: the accurate gate is slower, roughly 1.25x on `geglu` at large tensor sizes (the gate itself is ~2x, diluted by the surrounding slice and multiply). Small tensors are dispatch-bound and unaffected.

Fix proposed in #54902.
